controllers/nstemplatetier/nstemplatetier_controller_test.go

```diff
@@ -77,6 +83,256 @@ func TestReconcile(t *testing.T) {
+					require.NoError(t, err)
+					require.Len(t, ttrs.Items, len(tierTemplatesRefs)) // it's one TTR per each tiertemplate in the NSTemplateTier
+					t.Run("revision field is set but some TierTemplateRevision is missing, status should be updated", func(t *testing.T) {
+						// given
+						// the NSTemplateTier already has the revisions from previous test
+						// we delete the first TTR to make sure the status is updated and the TTR get's recreated
+						err = cl.Delete(context.TODO(), &ttrs.Items[0])
+						require.NoError(t, err)
+
+						// when
+						res, err = r.Reconcile(context.TODO(), req)
+						// then
+						require.NoError(t, err)
+						// revisions are the same
+						tiertest.AssertThatNSTemplateTier(t, "base1ns", cl).
+							HasStatusTierTemplateRevisions(tierTemplatesRefs)
+						// expected TierTemplateRevision CRs are there
+						ttrs := toolchainv1alpha1.TierTemplateRevisionList{}
+						err = cl.List(context.TODO(), &ttrs, runtimeclient.InNamespace(base1nsTier.GetNamespace()))
+						require.NoError(t, err)
+						require.Len(t, ttrs.Items, len(tierTemplatesRefs)) // it's one TTR per each tiertemplate in the NSTemplateTier
+					})
+				})
+
+			})
+
+			t.Run("revision field is set but TierTemplateRevision CRs are missing, they should be created", func(t *testing.T) {
+				// given
+				// the NSTemplateTier has already the status.revisions field populated
+				// but the TierTemplateRevision CRs are missing
+				tierTemplates := initTierTemplates(t, withTemplateObjects(crq), base1nsTier.Name)
+				base1nsTierWithRevisions := base1nsTier
+				base1nsTierWithRevisions.Status.Revisions = map[string]string{
+					"base1ns-admin-123456new": "base1ns-admin-123456new-abcd", "base1ns-clusterresources-123456new": "base1ns-clusterresources-123456new-abcd", "base1ns-code-123456new": "base1ns-code-123456new-abcd", "base1ns-dev-123456new": "base1ns-dev-123456new-abcd", "base1ns-edit-123456new": "`base1ns-edit-123456new-abcd", "base1ns-stage-123456new": "base1ns-stage-123456new-abcd", "base1ns-viewer-123456new": "base1ns-viewer-123456new-abcd"}
+				r, req, cl := prepareReconcile(t, base1nsTierWithRevisions.Name, append(tierTemplates, base1nsTierWithRevisions)...)
+				// when
+				// check no TTR is present before reconciling
+				tiertemplaterevision.AssertThatTTRs(t, cl, base1nsTierWithRevisions.GetNamespace()).DoNotExist()
+				_, err = r.Reconcile(context.TODO(), req)
+				// then
+				require.NoError(t, err)
+				// check that revisions field was populated
+				tiertest.AssertThatNSTemplateTier(t, "base1ns", cl).
+					HasStatusTierTemplateRevisions(tierTemplatesRefs)
+				// check that expected TierTemplateRevision CRs were created
+				ttrs := toolchainv1alpha1.TierTemplateRevisionList{}
+				err = cl.List(context.TODO(), &ttrs, runtimeclient.InNamespace(base1nsTierWithRevisions.GetNamespace()))
+				require.NoError(t, err)
+				require.Len(t, ttrs.Items, len(tierTemplatesRefs))
```

Comment:
correct me if I'm wrong, the TTR CRs should have different names than the one that were set in the `// given` part, right?
If so, could you please verify that here - that they are different?